src/SDK/Resource/Detectors/Host.php

```diff
@@ -52,8 +52,9 @@ private function getLinuxId(): ?string
         $paths = [self::PATH_ETC_MACHINEID, self::PATH_VAR_LIB_DBUS_MACHINEID];
 
         foreach ($paths as $path) {
-            if (file_exists($this->dir . $path)) {
-                return trim(file_get_contents($this->dir . $path));
+            $file = $this->dir . $path;
+            if (file_exists($file) && is_readable($file)) {
```

Comment:
```suggestion
            if (is_file($file) && is_readable($file)) {
```
Currently fails if `$file` is a readable directory.

Alternatively use only `file_get_contents()` and check return value to handle all possible failure scenarios:
```php
set_error_handler(static fn() => true);
try {
    if (($content = file_get_contents($file)) !== false) {
        return trim($content);
    }
} finally {
    restore_error_handler();
}
```